packages/cozy-harvest-lib/src/models/cipherUtils.spec.js

```diff
@@ -32,79 +19,137 @@ describe('createOrUpdateCipher', () => {
+            isLocked: jest.fn().mockResolvedValue(false)
+          }
+        })
+
+        vaultClient.getByIdOrSearch.mockResolvedValue(null)
+
+        const cipherId = null
+
+        const cipher = await cipherUtils.createOrUpdateCipher(
+          vaultClient,
+          cipherId,
+          { userCredentials, account, konnector }
+        )
+
+        expect(cipher).toMatchSnapshot()
+      })
     })
-    const cipherId = null
-    const cipher = await createOrUpdateCipher(vaultClient, cipherId, {
-      konnector,
-      account,
-      userCredentials
+
+    describe('when a cipher exists with the same credentials', () => {
+      it('should update the cipher with given credentials', async () => {
+        const { konnector, account, userCredentials, vaultClient } = setup({
+          vaultClient: {
+            isLocked: jest.fn().mockResolvedValue(false)
+          }
+        })
+
+        vaultClient.getByIdOrSearch.mockResolvedValue({
+          collectionIds: ['cozy-org-collection-id'],
+          id: 'existing-cipher',
+          login: {
+            password: 'password-to-be-updated',
+            uris: [
+              {
+                match: 'Domain',
+                uri: 'konnector-vendor-link'
+              }
+            ],
+            username: 'login-to-be-updated'
+          },
+          name: 'konnector-name',
+          organizationId: 'cozy-org-id',
+          type: 'Login'
+        })
+
+        const cipherId = null
+
+        const cipher = await cipherUtils.createOrUpdateCipher(
+          vaultClient,
+          cipherId,
+          { userCredentials, account, konnector }
+        )
+
+        expect(cipher).toMatchSnapshot()
+      })
     })
-    expect(vaultClient.getByIdOrSearch).toHaveBeenCalledTimes(2)
-    expect(vaultClient.getByIdOrSearch).toHaveBeenCalledWith(
-      'cipher-relationship-id',
-      {
-        type: 'Login',
-        uri: 'konnector-vendor-link',
-        username: 'my-login-credential'
-      },
-      [expect.any(Function), 'revisionDate']
-    )
-    expect(vaultClient.getByIdOrSearch).toHaveBeenCalledWith('found-cipher-id')
-    expect(cipher).toEqual({
-      id: 'saved-cipher-id'
+  })
+
+  describe('when given a cipherId', () => {
+    it('should update the cipher with given credentials', async () => {
+      const { konnector, account, userCredentials, vaultClient } = setup({
+        vaultClient: {
+          isLocked: jest.fn().mockResolvedValue(false)
+        }
+      })
+
+      vaultClient.getByIdOrSearch.mockResolvedValue({
+        collectionIds: ['cozy-org-collection-id'],
+        id: 'existing-cipher',
+        login: {
+          password: 'password-to-be-updated',
+          uris: [
+            {
+              match: 'Domain',
+              uri: 'konnector-vendor-link'
+            }
+          ],
+          username: 'login-to-be-updated'
+        },
+        name: 'konnector-name',
+        organizationId: 'cozy-org-id',
+        type: 'Login'
+      })
```

Comment:
Seems this is the same object as in above tests. This could be extracted and reused here. This would make it easier to understand what changes from test to test (I had to compare key by key this object with the other one above otherwise).

Reply:
I put it in a `existingCipher` variable, so it's clearer :+1: See https://github.com/cozy/cozy-libs/commit/3d2117605f016a81e9ae79e58937c4b91abb6516